### PR TITLE
Refactoring for user config

### PR DESCRIPTION
**Pull request by archanks**
thoughts?

### CODE REVIEW

File: js/ubar/config.js [hunk 1]
```diff
@@ -16,29 +16,21 @@
...
-    * settings. She has this amount of time do so before
-    * being redirected again. 
-    *
-    */
-    var expiration_time = {
-        disabled: moment.duration(2, 'weeks'),
-        enabled: moment.duration(1, 'year'),
-        redirected: moment.duration(60, 'seconds')
-      },
+    var defaultConfig = {
```
Comment (GregM) [inline js/ubar/config.js]: i still think we need to describe each of these things.  I think a defaultConfig is way too big to consume.  We should separate these by concerns.

File: js/ubar/storage.js [hunk 2]
```diff
@@ -17,13 +17,9 @@
     'use strict';
 
   var
-    UBAR_KEY = 'ubar',
+    UBAR_KEY        = 'ubar',
```
Comment (GregM) [inline js/ubar/storage.js]: tab or something

File: js/ubar/dom.js [hunk 3]
```diff
@@ -6,8 +6,8 @@
       'ubar_dom',
```
Comment (GregM) [inline js/ubar/dom.js]: this file is good. +1

File: js/ubar/storage.js [hunk 4]
```diff
@@ -17,13 +17,9 @@
...
     REDIRECTED_NAME = 'ubar_redirected',
```
Comment (GregM) [inline js/ubar/storage.js]: otherwise the file is good. +1

File: js/ubar/ubar.js [hunk 5]
```diff
@@ -2,13 +2,24 @@
 
   if (typeof define === 'function' && define.amd) {
     // AMD. Register as an anonymous module.
-    define('ubar', ['config.js','../node_modules/bean/bean.min.js', '../node_modules/when/when.js' ], factory);
+    define(
+      'ubar',
+
+      [
+      'config.js',
+      '../node_modules/bean/bean.min.js',
```
Comment (GregM) [inline js/ubar/ubar.js]: still need to figure this out. probably in require.config

File: js/ubar/ubar.js [hunk 6]
```diff
@@ -42,14 +40,14 @@
      */
 
     function bindOnBannerButtonEvents () {
-      var ubarComponentDiv = document.querySelectorAll('.' + UBAR_COMPONENT_CLASS)[0],
-          onButton = ubarComponentDiv.querySelectorAll('.' + UBAR_BUTTON_CLASSES.TURN_ON)[0],
-          installAppButton = ubarComponentDiv.querySelectorAll('.' + UBAR_BUTTON_CLASSES.INSTALL_APP)[0],
-          closeBannerButton = ubarComponentDiv.querySelectorAll('.' + UBAR_BUTTON_CLASSES.CLOSE_BANNER)[0];
+      var ubarComponentDiv    = document.querySelectorAll('.' + (USER_CONFIG.component_class || DEFAULT_CONFIG.component_class) )[0],
```
Comment (GregM) [inline js/ubar/ubar.js]: how are we imagining USER_CONFIG gets set?

Comment (GregM) [inline js/ubar/ubar.js]: I feel like we should say in the top:

var ubarConfig = DEFAULT_CONFIG || USER_CONFIG;

OR

var ubarConfig= DEFAULT_CONFIG || {};

File: js/ubar/ubar.js [hunk 7]
```diff
@@ -219,25 +217,59 @@
...
+      var timeString = time_string.split(" "),
+          timeValue  = parseInt(timeString[0], 10);
+          timeUnit   = timeString[1];
+
+      return moment.duration( timeValue, timeUnit ).asSeconds() ;
+    }
+
+    /* Initialize UBAR with following params:
+     * {
+     *   ios_app_store_url       : 'https://itunes.apple.com/us/app/appname/id331804452?mt=8',
```
Comment (GregM) [inline js/ubar/ubar.js]: i don't think we want to put the values in the comments. it's going to be tough to maintain. we should just direct people to where they can see the default values

File: js/ubar/ubar.js [hunk 8]
```diff
@@ -219,25 +217,59 @@
...
     }
 
-    function init (appStoreLink, host, path) {
+    /**
+     * Get Time in Seconds
+     *
+     * @private
+     * @method getTimeinSeconds
+     */
+    function getTimeinSeconds(time_string) {
```
Comment (sanchezjjose) [inline js/ubar/ubar.js]: Should we make this a generic getTime function, and just return the duration object? This way, the client can decide whether to use seconds, milliseconds, etc...

File: js/ubar/ubar.js (hunk 6 above)
Comment (archanks) [inline js/ubar/ubar.js]: USER_CONFIG gets passed into init and gets set there. (lines 261 - 263)

File: js/ubar/ubar.js (hunk 5 above)
Comment (archanks) [inline js/ubar/ubar.js]: do you mean the relative url to the js file ?

File: js/ubar/storage.js [hunk 9]
```diff
@@ -84,7 +80,7 @@
    * @public
    * @method setRedirected
    */
-  function setRedirected () {
+  function setRedirected (EXP_DURATION_REDIRECTED_MS) {
```
Comment (sanchezjjose) [inline js/ubar/storage.js]: Here we are passing in a variable which assumes Milliseconds, but if you look at the configuration value coming in (manage_window_time) in the defaultConfig, it says 60 seconds. We should look at how we can abstract the time unit better.

File: js/ubar/config.js (hunk 1 above)
Comment (archanks) [inline js/ubar/config.js]: I agree. Not sure where would be best to explain this. 
Howe about if we explained all the values in the defaultConfig inside the config file ?

File: js/ubar/storage.js (hunk 2 above)
Comment (archanks) [inline js/ubar/storage.js]: hah i think this was my sublime alignment plugin adding spaces so the equal to signs of the variables UBAR_KEY and REDIRECTED_NAME match up

File: js/ubar/ubar.js (hunk 5 above)
Comment (GregM) [inline js/ubar/ubar.js]: yep

File: js/ubar/storage.js (hunk 9 above)
Comment (archanks) [inline js/ubar/storage.js]: we could either pass the moment object into this and change it to seconds inside the function, for the purposes of setting the cookie expiration

File: js/ubar/ubar.js (hunk 8 above)
Comment (archanks) [inline js/ubar/ubar.js]: so it would return the moment object ?
i think that sounds great!

File: js/ubar/storage.js (hunk 2 above)
Comment (GregM) [inline js/ubar/storage.js]: ahh got it :)

File: js/ubar/config.js (hunk 1 above)
Comment (GregM) [inline js/ubar/config.js]: Yes, let me take a quick pass at this now

File: js/ubar/ubar.js (hunk 7 above)
Comment (archanks) [inline js/ubar/ubar.js]: that sounds good... so something that says see config.js for all values and in config.js explain all the keys and what their values mean ?

File: js/ubar/ubar.js (hunk 8 above)
Comment (sanchezjjose) [inline js/ubar/ubar.js]: sweet, i'll make the change here on github :) just learned you could do that.

**Pull request merged**